src/eos-prepare-volume.c

```diff
@@ -137,7 +137,11 @@ main (int argc,
     }
 
   sysroot = ostree_sysroot_new_default ();
-  if (!ostree_sysroot_load (sysroot, NULL, &error))
+
+  /* Lock the sysroot so it can’t be updated while we’re pulling from it. The
+   * lock is automatically released when we finalise the sysroot. */
+  if (!ostree_sysroot_lock (sysroot, &error) ||
```

Comment:
Just FYI, this will block indefinitely until the lock is acquired. I think that's completely fine right now, but I wanted to make sure you were aware of that. You can also use `ostree_sysroot_try_lock` for non-blocking that will fail immediately if the lock can't be acquired or `ostree_sysroot_lock_async` to have complete non-blocking control.